### PR TITLE
add interface PacketBoards for sending prepared boards to clients

### PR DESCRIPTION
**Pull request by ghost**
add interface PacketBoards for sending prepared boards to clients
add NetworkPacketBoard (implementation of PacketBoards) + tests
refactor PacketMove, now we can add list of integer instead of one position
add in main (client, server) handling packets with boards and movements

### CODE REVIEW

File: common/src/main/java/com/blackship/battlesheep/communication/packet/BasePacket.java [hunk 1]
```diff
@@ -11,6 +11,27 @@
...
+    }
+
+    @Override
+    public LocalTime getCreationTime() {
+        return time;
+    }
+
+    @Override
+    public Packet setCreationTime(LocalTime time) {
+        this.time = time;
```
Comment (arturAR) [inline common/src/main/java/com/blackship/battlesheep/communication/packet/BasePacket.java:28]: 👎

**Review by arturAR — APPROVED** [review]

File: client/src/main/java/com/blackship/battlesheep/communication/network/AppClientHandler.java [hunk 2]
```diff
@@ -3,26 +3,57 @@
...
+
+        log.info("...Reading from server...");
+        Packet receivedPacket = clientCommunicationHandler.read();
+        log.info("...Received: " + receivedPacket + "  " + receivedPacket.getCreationTime());
+
+        return this;
+    }
+
+    public AppClientHandler receiveBoards() throws IOException, ClassNotFoundException {
+        log.info("...Reading boards from server...");
```
Comment (Devas) [inline client/src/main/java/com/blackship/battlesheep/communication/network/AppClientHandler.java:53]: Nice logging :+1:

File: client/src/test/java/com/blackship/battlesheep/communication/network/AppClientSocketIT.java [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
  * @author milosz
  * @since 28.07.2017
  */
 public class AppClientSocketIT {
 
+    private byte[] readBytesFromServer(Socket givenClientSocket) throws IOException {
+        return new NetworkReader(
+                new NetworkStreams(givenClientSocket)
+                        .getInput()
+                        .orElseThrow(() -> new IOException("no inputStream !"))
```
Comment (Devas) [inline client/src/test/java/com/blackship/battlesheep/communication/network/AppClientSocketIT.java:22]: :+1:

File: client/src/test/java/com/blackship/battlesheep/communication/network/AppClientSocketIT.java [hunk 4]
```diff
@@ -1,23 +1,29 @@
...
+        ).read();
+
+    }
+
     @Test(timeOut = 2000)
```
Comment (Devas) [inline client/src/test/java/com/blackship/battlesheep/communication/network/AppClientSocketIT.java:27]: :+1:

**Review by Devas — APPROVED** [review]
Nice looking code

**Pull request merged**